From a667a0a5619156774fb688078597d05cd2eb7127 Mon Sep 17 00:00:00 2001
From: Ken Zhu <guigenz@codeaurora.org>
Date: Tue, 20 Mar 2018 12:18:01 -0700
Subject: [PATCH 239/281] linux: clean the cb before icmp_send

In the icmp_send function, skb->cb is used as ip_option, but it may be used
by other sub-systems before ipv4_link_failure.

So clean the skb->cb before entering icmp_send.

Change-Id: I985b363e7f1b00314843bcda1fa8b78bbc1bd7d8
Signed-off-by: Ken Zhu <guigenz@codeaurora.org>
---
 net/ipv4/route.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/net/ipv4/route.c b/net/ipv4/route.c
index 160c8814b746..7237eaa16481 100644
--- a/net/ipv4/route.c
+++ b/net/ipv4/route.c
@@ -1246,6 +1246,9 @@ static void ipv4_link_failure(struct sk_buff *skb)
 {
 	struct rtable *rt;
 
+	/* Forwarding packets, do not have IPCB() initialized, do so
+	 */
+	memset(IPCB(skb), 0, sizeof(struct inet_skb_parm));
 	ipv4_send_dest_unreach(skb);
 
 	rt = skb_rtable(skb);
-- 
2.17.1

